applications/iio/cpp/pluto_fft_example/support_operators.hpp

```diff
@@ -0,0 +1,731 @@
+               burst_size * sizeof(complex),
+               cudaMemcpyDeviceToHost);
+
+    // Convert to magnitude spectrum using common utility function
+    auto magnitude_spectrum =
+        fft_utils::convertToMagnitudeSpectrum(host_data, burst_size, power_offset_.get());
+
+    // Write data file for gnuplot with frequency axis from -fs/2 to +fs/2
+    std::string data_file = output_file_.get() + ".dat";
+    std::ofstream data_stream(data_file);
+
+    float freq_step = max_frequency_.get() / static_cast<float>(burst_size);
+    float freq_step_mhz = freq_step / 1e6f;  // Convert to MHz
+
+    for (size_t i = 0; i < burst_size; ++i) {
+      // Map frequency axis to [-fs/2, +fs/2] range in MHz
+      float frequency_mhz =
+          (static_cast<float>(i) - static_cast<float>(burst_size) / 2.0f) * freq_step_mhz;
+      data_stream << frequency_mhz << " " << magnitude_spectrum[i] << std::endl;
+    }
+    data_stream.close();
+
+    // Find peak frequency for title using common utility function
+    auto [peak_freq_mhz, peak_magnitude] =
+        fft_utils::findPeakFrequency(magnitude_spectrum, burst_size, freq_step_mhz);
+
+    // Create gnuplot script
+    std::string script_file = output_file_.get() + ".gp";
+    std::ofstream script_stream(script_file);
+
+    script_stream << "set terminal png size 1200,800\n";
+    script_stream << "set output '" << output_file_.get() << ".png'\n";
+    script_stream << "set title 'Pluto SDR FFT Spectrum - Peak: " << std::fixed
+                  << std::setprecision(2) << peak_freq_mhz << " MHz (" << std::setprecision(1)
+                  << peak_magnitude << " dB)'\n";
+    script_stream << "set xlabel 'Frequency (MHz)'\n";
+    script_stream << "set ylabel 'Magnitude (dB)'\n";
+
+    script_stream << "set grid\n";
+    script_stream << "set style line 1 linecolor rgb '#00ff00' linewidth 2\n";
+    script_stream << "plot '" << data_file << "' with lines linestyle 1 title 'FFT Magnitude'\n";
+    script_stream.close();
+
+    // Execute gnuplot
+    std::string gnuplot_cmd = "gnuplot " + script_file;
+    int result = std::system(gnuplot_cmd.c_str());
+
+    if (result == 0) {
+      HOLOSCAN_LOG_INFO("Gnuplot spectrum saved to: {}.png", output_file_.get());
+      HOLOSCAN_LOG_INFO(
+          "Peak frequency: {:.2f} MHz with magnitude: {:.1f} dB", peak_freq_mhz, peak_magnitude);
+    } else {
+      HOLOSCAN_LOG_ERROR("Gnuplot execution failed with return code: {}", result);
+    }
+
+    // Display the plot using system image viewer (optional)
+    std::string display_cmd = "xdg-open " + output_file_.get() + ".png &";
+    std::system(display_cmd.c_str());
+
+    // Clean up temporary files
+    std::remove(data_file.c_str());
+    std::remove(script_file.c_str());
+
+    HOLOSCAN_LOG_INFO("FFT gnuplot visualization complete. Application will now exit.");
+
+    // Request application shutdown after processing one spectrum
+    std::exit(0);
```

Comment:
Calling `exit()` in `compute()` might be a bit awkward. Can you instead use a `CountCondition` so that it only runs once and the application naturally exits? Something like this:
https://github.com/nvidia-holoscan/holoscan-sdk/blob/main/examples/ping_simple/cpp/ping_simple.cpp#L27